linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -5499,6 +5499,12 @@ private[optimizer] object OptimizerCore {
         private[FreshNameAllocator] val usedNamesToNextCounter: Map[N, Int])
   }
 
+  def originalNameForFresh(base: Name, originalName: OriginalName,
+      freshName: Name): OriginalName = {
+    if (originalName.isDefined || (freshName eq base)) originalName
+    else OriginalName(base)
+  }
+
```

Comment:
Providing originalName handling as a separate method seems to make it easy to get it wrong: Since original names are only at definitions sites now, every time one requests a new fresh name, one needs to deal with the original name (and only then non-trivial handling is required). Have you considered moving this into the fresh methods directly?

Reply:
That's a very good point! I changed it in `OptimizerCore`, but I'm not sure where else you think I should do that. In the compiler backend, the `fresh` methods are used only for synthetics and from `encodeLocalSym`, and for the latter we have the `originalNameOfLocal` which forces to do the right thing.

Reply:
Right, in the compiler backend that does not apply.